peer/connmgr.go

```diff
@@ -0,0 +1,369 @@
+package peer
```

Comment:
Note: I will still add unit tests for this. Just waiting for a go-ahead on the approach first